plugins/inputs/mqtt_consumer/mqtt_consumer.go

```diff
@@ -186,18 +207,28 @@ func (m *MQTTConsumer) recvMessage(_ mqtt.Client, msg mqtt.Message) {
 func (m *MQTTConsumer) Stop() {
 	m.Lock()
 	defer m.Unlock()
-	close(m.done)
-	m.client.Disconnect(200)
-	m.started = false
+
+	if m.started {
```

Comment:
Rename started field to connected, since we have changed the meaning.

Reply:
Will do, Thanks!